### PR TITLE
Add Plugin Support

### PR DESCRIPTION
**Pull request by theduckylittle**
This might _never_ get merged or expect the work to live on elsewhere.

This is an example of how one might build a react-based plugin for GeoMoose.

### CODE REVIEW

**Comment by elil** [comment]
This is a great way to show by example some of the possibilities!

**Comment by klassenjs** [comment]
The `window.React = React` stuff scares me.

**Comment by klassenjs** [comment]
But as long as we are discussing dirty hacks, can we put `React` in `gm3.Application`?

At least the `app` global is the user's discretion and doesn't magically pollute window.

And the circular references don't matter if the object is never deleted, right?

**Comment by theduckylittle** [comment]
@klassenjs That has also been a concern of mine. I just refactored the code a little to _pass_ react in as a prop to the component. It requires the user do a little extra work but really would allow anyone with the motivation to really push a GeoMoose integration.

**Comment by theduckylittle** [comment]
In terms of references, React will never get unloaded because of how we use it in the app.

**Comment by theduckylittle** [comment]
@klassenjs -  I realized I didn't actually push my changes. Now you can see the updated version. I think this satisfies everyone's scope concerns.

**Comment by klassenjs** [comment]
How does hello.js get to examples/desktop/hello.js?

**Comment by theduckylittle** [comment]
I would like to change this to get rid of the change to examples directory. I can write up the how-to if you're okay with the implementation.

**Comment by theduckylittle** [comment]
Updated the docs, cleaned up the changes to the desktop example

File: docs/howto/add-a-plugin.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+2. The following will add a new tab to the list of tabs:
+
+   ```
+   <span id="hello-tab-tab" class="tab" onclick="showTab('hello-tab', event)">Hello Tab</span>
+   ```
+
+3. And add the new tab content:
+
+    ```
+    <div class="tab-content" id="service-tab"></div>
```
Comment (klassenjs) [inline docs/howto/add-a-plugin.md:53]: id="hello-tab"

File: docs/howto/add-a-plugin.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ```
+
+### Adding the plug-in
+
+1. The plug-in will need copied from where it was just built to your application's directory. In the demo this is `examples/desktop` but your apps directory may be different:
+
+    ```
+    cp examples/plugins/hello.js examples/desktop/
```
Comment (klassenjs) [inline docs/howto/add-a-plugin.md:61]: plugin not plugins

File: docs/howto/add-a-plugin.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+deeper level. They require familiarity with React and Redux
+but expose many of GeoMoose's inner-workings.
+
+## 1. Write the plug-in
+
+Plugins are given the React, ReactDOM, and redux store.
+They are expected to return a component. The following example
+uses a functional component:
+
+`examples/plugins/src/hello.js`:
```
Comment (klassenjs) [inline docs/howto/add-a-plugin.md:13]: plugin not plugins

File: docs/howto/add-a-plugin.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    </div>
+);
+```
+
+This example is written using ES6. It features the JSX syntax. It is not
+the scope of this How-to describe those concepts, however, the GeoMoose repository
+contains all the things we need to convert this to browser-friendly ES6.
+
+```
+cd examples/plugins
```
Comment (klassenjs) [inline docs/howto/add-a-plugin.md:29]: plugin not plugins

File: docs/howto/add-a-plugin.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+npm install
+npm run build
+```
+
+This will create `examples/plugins/hello.js`.
```
Comment (klassenjs) [inline docs/howto/add-a-plugin.md:34]: plugin not plugins

File: docs/howto/add-a-plugin.md (hunk 1 above)
Comment (klassenjs) [reply docs/howto/add-a-plugin.md:53]: And maybe show the before and after state.

**Comment by theduckylittle** [comment]
Fixed the editing issues!

**Pull request merged**